runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsPlugin.java

```diff
@@ -271,4 +308,28 @@ else if (event.getGroupId() == WidgetID.BARROWS_PUZZLE_GROUP_ID)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onLocalPlayerDeath(LocalPlayerDeath event)
+	{
+		stopPrayerDrainTimer();
```

Comment:
There are more ways to leave than just dieing, and i think this would trigger the gamestate change from above, so why is this here?

Reply:
Oh, if a a gamestate change happens on death, then I probably don't need this.

Reply:
I think it does. Why did you add it? You should be testing whether or not the plugin behaves correctly when leaving or teleporting from the crypt.

Reply:
I think I just added it because I saw it in another plugin. But you're right, the existing game state change logic catches all those cases.